react/features/shared-video/components/native/AbstractVideoManager.js

```diff
@@ -0,0 +1,288 @@
+
+        if (_isOwner) {
+            return;
+        }
+
+        const playerTime = await this.getTime();
+
+        if (shouldSeekToPosition(_time, playerTime)) {
+            this.seek(_time);
+        }
+
+        if (this.getPlaybackState() !== _status) {
+            if (_status === PLAYBACK_STATES.PLAYING) {
+                this.play();
+            }
+
+            if (_status === PLAYBACK_STATES.PAUSED) {
+                this.pause();
+            }
+        }
+    }
+
+    /**
+     * Handle video playing.
+     *
+     * @returns {void}
+     */
+    onPlay() {
+        this.fireUpdateSharedVideoEvent();
+    }
+
+    /**
+     * Handle video paused.
+     *
+     * @returns {void}
+     */
+    onPause() {
+        this.fireUpdateSharedVideoEvent();
+    }
+
+    /**
+     * Dispatches an update action for the shared video.
+     *
+     * @returns {void}
+     */
+    async fireUpdateSharedVideoEvent() {
+        const { _isOwner } = this.props;
+
+        if (!_isOwner) {
+            return;
+        }
+
+        const status = this.getPlaybackState();
+
+        if (!Object.values(PLAYBACK_STATES).includes(status)) {
+            return;
+        }
+
+        const time = await this.getTime();
+
+        const {
+            _ownerId,
+            _setSharedVideoStatus,
+            _videoUrl
+        } = this.props;
+
+        _setSharedVideoStatus({
+            videoUrl: _videoUrl,
+            status,
+            time,
+            ownerId: _ownerId
+        });
+    }
+
+    /**
+     * Seeks video to provided time
+     * @param {number} time
+     */
+    seek: (time: number) => void;
+
+    /**
+     * Indicates the playback state of the video
+     */
+    getPlaybackState: () => boolean;
+
+    /**
+     * Plays video
+     */
+    play: () => void;
+
+    /**
+     * Pauses video
+     */
+    pause: () => void;
+
+    /**
+     * Retrieves current time
+     */
+    getTime: () => number;
+
+    /**
+     * Disposes current video player
+     */
+    dispose: () => void;
+}
+
+
+export default AbstractVideoManager;
+
+/**
+ * Maps part of the Redux store to the props of this component.
+ *
+ * @param {Object} state - The Redux state.
+ * @returns {Props}
+ */
+export function _mapStateToProps(state: Object): $Shape<Props> {
+    const { ownerId, status, time, videoUrl } = state['features/shared-video'];
+    const localParticipant = getLocalParticipant(state);
+
+    return {
+        _conference: getCurrentConference(state),
+        _isOwner: ownerId === localParticipant.id,
+        _ownerId: ownerId,
+        _status: status,
+        _time: time,
+        _videoUrl: videoUrl
+    };
+}
+
+/**
+ * Maps part of the props of this component to Redux actions.
+ *
+ * @param {Function} dispatch - The Redux dispatch function.
+ * @returns {Props}
+ */
+export function _mapDispatchToProps(dispatch: Function): $Shape<Props> {
+    return {
+        _setSharedVideoStatus: ({ videoUrl, status, time, ownerId }) => {
```

Comment:
Generally, we've found `mapDispatchToProps` not to be very useful. It's a lot of boilerplate for a one liner function call here. Do you mind just dispatch()-ing the action directly instead?